### PR TITLE
Adding proper validation for PropType oneOf and OneOfType.

### PR DESCRIPTION
**Pull request by alansouzati**
I had people on my team sending wrong arguments to oneOf and oneOfType attributes from PropTypes.

The current message exposed some internal attributes to the user with the following message:

```bash
Warning: Failed propType: checker is not a function Check the render method of Chart.
```

This warning is a little misleading, I had to spend some time trying to figure out who was invoking this **checker** function, unsuccessfully. Then, after some time, I realized the argument to oneOf and oneOfType was actually invalid and it had nothing to do with checker function.

I've opened a [StackOverflow](http://stackoverflow.com/questions/29995444/react-checker-is-not-a-function) case for this.

This pull request aims to fix this problem by validating the argument type as an array in both createUnionTypeChecker and createEnumTypeChecker. I've added the corresponding tests for this.

Alan

### CODE REVIEW

File: src/classic/types/__tests__/ReactPropTypes-test.js [hunk 1]
```diff
@@ -520,6 +520,12 @@ describe('ReactPropTypes', function() {
   });
 
   describe('OneOf Types', function() {
+  	it("should fail for invalid argument", function() {
+  		var error = PropTypes.oneOf('red', 'blue');
```
Comment (zpao) [inline src/classic/types/__tests__/ReactPropTypes-test.js]: We don't have lint properly running in test files yet, but can you make sure these are converted to 2 spaces as well

Comment (alansouzati) [inline src/classic/types/__tests__/ReactPropTypes-test.js]: oh yea, working on it!

**Comment by alansouzati** [comment]
Hey, I've added proper formatting for the test files, could you please double-check?

**Comment by zpao** [comment]
@alansouzati Can you squash and rebase? We moved a bunch of files around recently so this doesn't apply cleanly (and would be good to have a single commit)

**Comment by alansouzati** [comment]
Sure I will work on that and will send a new pull request.

**Comment by zpao** [comment]
You can do that in the same branch, you don't need to do a new PR if you don't want to.

**Comment by alansouzati** [comment]
Done, let's wait to hear from travis :smile:

**Comment by alansouzati** [comment]
Hey any idea why it is failing remotely but is not failing locally? And whatever the failing scenarios are, I believe it is not related to what I've done. Any thoughts appreciated.

File: src/isomorphic/classic/types/ReactPropTypes.js [hunk 2]
```diff
@@ -249,7 +255,13 @@ function createObjectOfTypeChecker(typeChecker) {
...
 
 function createUnionTypeChecker(arrayOfTypeCheckers) {
-  function validate(props, propName, componentName, location, propFullName) {
+  if (!Array.isArray(arrayOfTypeCheckers)) {
+    return new Error(
+      `Invalid argument supplied to oneOfType, expected an instance of array.`
+    );
+  }
+
+  function validate(props, propName, componentName, location) {
```
Comment (zpao) [inline src/isomorphic/classic/types/ReactPropTypes.js]: The lint error is because you inadvertently removed a param here (`propFullName`)

**Comment by alansouzati** [comment]
It is true, my fault. Sorry about that!

**Comment by alansouzati** [comment]
Hey, I did the required changes. Just to make sure, is there anything left on my end?

**Comment by zpao** [comment]
Can you squash the commits into a single commit?

**Comment by alansouzati** [comment]
Im having a hard time squashing this PR. I will create a new one.

**Pull request closed without merging**